docs/faq.md

```diff
@@ -142,3 +142,23 @@ Refer to the [PSPDFKit blog post for more details.](https://pspdfkit.com/blog/20
 ```swift
 UIApplication.sharedApplication().keyWindow?.layer.speed = 100
 ```
+
+**How do I reset application state before each test using Swift?**
+
+In the application's Build Settings, set **Defines Module** to **Yes**. Create a `resetApplication()` method on the AppDelegate.
+
+In the EarlGrey test target, import the application using `@testable`. In `setUp()`, aquire a reference to the delegate then invoke `resetApplication()`.
+
+```swift
+// swift
+@testable import App
+
+class MyTests: XCTestCase {
+
+    override func setUp() {
```

Comment:
In what scenarios would the previous run not clean up the state? Are you worried about the app crashing or the same device previously having the app installed?

In a clean simulator scenario, when a test is run for the first time, shouldn't the app be in a clean state? So then, in setUp method you would:
1. assert the state is clean
2. setup the required mocks and fake data
3. Create other test fixtures for the app.

Then test case begins, modifies the state of the app and in tearDown, the app state is cleaned again, mocks are removed, fake data is reset, etc. And the cycle repeats...